Deal with temp tables after test shutdown

We have tests that compare set of file after compute postgres shutdown with the set of files that comes from the pageserver through `basebackup` command.

We stop compute node in fast mode, so files from temp tables would not be cleaned up on exit. And test routine sometimes can catch that.

https://app.circleci.com/pipelines/github/zenithdb/zenith/3574/workflows/88acf4a3-b206-4140-ac08-b8e0da56b808/jobs/30497 

```
AssertionError: assert ['./PG_VERSIO...ode.map', ...] == ['./PG_VERSIO...ode.map', ...]
  At index 12 diff: 'base/16384/t16_31039' != 'global/PG_VERSION'
  Left contains 2 more items, first extra item: 'pg_multixact/offsets/0000'
  Full diff:
    [
     './PG_VERSION',
     './pg_hba.conf',
     './pg_ident.conf',
     './postgresql.auto.conf',
     'base/1/PG_VERSION',
     'base/1/pg_filenode.map',
     'base/12973/PG_VERSION',
     'base/12973/pg_filenode.map',
     'base/12974/PG_VERSION',
     'base/12974/pg_filenode.map',
     'base/16384/PG_VERSION',
     'base/16384/pg_filenode.map',
  +  'base/16384/t16_31039',
  +  'base/16384/t16_31041',
     'global/PG_VERSION',
     'global/pg_filenode.map',
     'pg_multixact/members/0000',
     'pg_multixact/offsets/0000',
     'pg_xact/0000',
    ]
test_runner/batch_pg_regress/test_pg_regress.py:58: in test_pg_regress
    check_restored_datadir_content(test_output_dir, env, pg)
test_runner/fixtures/zenith_fixtures.py:1246: in check_restored_datadir_content
    assert pgdata_files == restored_files
E   AssertionError: assert ['./PG_VERSIO...ode.map', ...] == ['./PG_VERSIO...ode.map', ...]
E     At index 12 diff: 'base/16384/t16_31039' != 'global/PG_VERSION'
E     Left contains 2 more items, first extra item: 'pg_multixact/offsets/0000'
E     Full diff:
E       [
E        './PG_VERSION',
E        './pg_hba.conf',
E        './pg_ident.conf',
E        './postgresql.auto.conf',
E        'base/1/PG_VERSION',
E        'base/1/pg_filenode.map',
E        'base/12973/PG_VERSION',
E        'base/12973/pg_filenode.map',
E        'base/12974/PG_VERSION',
E        'base/12974/pg_filenode.map',
E        'base/16384/PG_VERSION',
E        'base/16384/pg_filenode.map',
E     +  'base/16384/t16_31039',
E     +  'base/16384/t16_31041',
E        'global/PG_VERSION',
E        'global/pg_filenode.map',
E        'pg_multixact/members/0000',
E        'pg_multixact/offsets/0000',
E        'pg_xact/0000',
E       ]
```

--- a/test_runner/fixtures/zenith_fixtures.py
+++ b/test_runner/fixtures/zenith_fixtures.py
@@ -1170,6 +1170,40 @@ def test_output_dir(request: Any) -> str:
     return test_dir
 
 
+SKIP_DIRS = frozenset(('pg_wal', 'pg_stat', 'pg_stat_tmp', 'pg_subtrans', 'pg_logical'))
+
+SKIP_FILES = frozenset(('pg_internal.init',
+                        'pg.log',
+                        'zenith.signal',
+                        'postgresql.conf',
+                        'postmaster.opts',
+                        'postmaster.pid',
+                        'pg_control'))
+
+
+def should_skip_dir(dirname: str) -> bool:
+    return dirname in SKIP_DIRS
+
+
+def should_skip_file(filename: str) -> bool:
+    if filename in SKIP_FILES:
+        return True
+    # check for temp table files according to https://www.postgresql.org/docs/current/storage-file-layout.html
+    # i e "tBBB_FFF"
+    if not filename.startswith('t'):
+        return False
+
+    tmp_name = filename[1:].split('_')
+    if len(tmp_name) != 2:
+        return False
+
+    try:
+        list(map(int, tmp_name))
+    except:
+        return False
+    return True
+
+
 #
 # Test helpers
 #
@@ -1179,19 +1213,10 @@ def list_files_to_compare(pgdata_dir: str):
         for filename in filenames:
             rel_dir = os.path.relpath(root, pgdata_dir)
             # Skip some dirs and files we don't want to compare
-            skip_dirs = ['pg_wal', 'pg_stat', 'pg_stat_tmp', 'pg_subtrans', 'pg_logical']
-            skip_files = [
-                'pg_internal.init',
-                'pg.log',
-                'zenith.signal',
-                'postgresql.conf',
-                'postmaster.opts',
-                'postmaster.pid',
-                'pg_control'
-            ]
-            if rel_dir not in skip_dirs and filename not in skip_files:
-                rel_file = os.path.join(rel_dir, filename)
-                pgdata_files.append(rel_file)
+            if should_skip_dir(rel_dir) or should_skip_file(filename):
+                continue
+            rel_file = os.path.join(rel_dir, filename)
+            pgdata_files.append(rel_file)
 
     pgdata_files.sort()
     log.info(pgdata_files)

